### PR TITLE
Back-End: Board Of Appeals letter support for CST DDL

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- *Added Board Of Appeals letter support for CST DDL and put it behind a feature flag.*

## Related issue(s)
- *[Link to ticket created in va.gov-team repo](https://github.com/department-of-veterans-affairs/va.gov-team/issues/55272)*
- *Related PR: https://github.com/department-of-veterans-affairs/vets-website/pull/23835*

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected*
- *Describe the tests completed and the results*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
Impacts the Claims Status Tool Down Decision Letters feature.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/claim_letters/claim_letter_downloader.rb [hunk 1]
```diff
@@ -7,7 +7,13 @@ class ClaimLetterDownloader
     FILENAME = 'ClaimLetter'
     # 184: Notification Letter (e.g. VA 20-8993, VA 21-0290, PCGL)
     # 339: Rating Decision Letter
-    DOC_TYPE_ALLOWLIST = %w[184].freeze
+    # 27: Board Of Appeals Decision Letter
+    DOC_TYPE_ALLOWLIST =
+      if Flipper.enabled?(:cst_include_ddl_boa_letters, @current_user)
```
Comment (kjduensing) [inline lib/claim_letters/claim_letter_downloader.rb:12]: I was thinking about this - do we even need a feature flag here? This seems like an incremental update to the existing DDL feature that already exists. It adds complexity and the technical risk is minimal.

Tagging @jacobworrell @jerekshoe @samcoforma for input. Thoughts?

**Review by aherzberg — APPROVED** [review]
LGTM for the Mobile team

File: lib/claim_letters/claim_letter_downloader.rb (hunk 1 above)
Comment (samcoforma) [reply lib/claim_letters/claim_letter_downloader.rb:12]: @kjduensing per our convo, could I get an approval on this?

Comment (kjduensing) [reply lib/claim_letters/claim_letter_downloader.rb:12]: I think we answered this indirectly through yesterday's demo: we're not entirely sure what the PDF version of the letter contains, so we want to hide this behind a feature flag to verify

**Review by kjduensing — APPROVED** [review]
Looks good to me 🇺🇸

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**